src/vs/vscode.d.ts

```diff
@@ -4560,6 +4589,13 @@ declare module 'vscode' {
 		 */
 		readonly processId: Thenable<number>;
 
+		/**
+		 * The current dimensions of the terminal. This will be `undefined` immediately after the
+		 * terminal is created as the dimensions are not known until shortly after the terminal is
+		 * created.
+		 */
+		readonly dimensions: TerminalDimensions | undefined;
```

Comment:
@jrieken we spoke earlier about how this could probably not be undefined. I've changed my mind because while we could know the dimensions immediately for terminals created from the renderer process, terminals and terminal renderers created on the ext host have no way to know about their dimensions. When the ext host supports splitting this will be even more difficult/impossible. I think it's better to just say we don't know initially and fire an event shortly after.